net/davix/src/TDavixFile.cxx

```diff
@@ -641,6 +642,7 @@ Int_t TDavixFileInternal::DavixStat(const char *url, struct stat *st)
 {
    DavixError *davixErr = NULL;
 
+   TLockGuard guard(&davixLock);
    if (davixPosix->stat(davixParam, url, st, &davixErr) < 0) {
```

Comment:
```suggestion
   int stat_result = -1;
   { 
      TLockGuard guard(&davixLock);
      stat = davixPosix->stat(davixParam, url, st, &davixErr);
   }
   if ( stat_result < 0) {
```
because of the error message.